pkg/multus/multus.go

```diff
@@ -369,11 +369,14 @@ func DelegateAdd(exec invoke.Exec, kubeClient *k8s.ClientInfo, pod *v1.Pod, dele
 	}
 
 	if pod != nil {
-		// send kubernetes events
-		if delegate.Name != "" {
-			kubeClient.Eventf(pod, v1.EventTypeNormal, "AddedInterface", "Add %s %v from %s", rt.IfName, ips, delegate.Name)
-		} else {
-			kubeClient.Eventf(pod, v1.EventTypeNormal, "AddedInterface", "Add %s %v", rt.IfName, ips)
+		// check Interfaces and IPs because some CNI plugin just return empty result
+		if res.Interfaces != nil || res.IPs != nil {
```

Comment:
checking interfaces should be enough? In which case the interfaces is equal to nil and IPs not nil?

Reply:
In case of CNI 0.2.0, there is no 'interface' in result object, so we also need to double check IPs as well.

Reply:
but what happens if you run a regular cni like bridge version 0.2.0 without IPAM?

Reply:
0.2.0 does require IPAM. Please see CNI 0.2.0 SPEC.